lib/rubocop/cop/rspec/metadata_style.rb

```diff
@@ -45,6 +45,11 @@ class MetadataStyle < Base # rubocop:disable Metrics/ClassLength
         PATTERN
 
         def on_metadata(symbols, hash)
+          # RSpec example groups accept two string arguments. In such a case,
+          # the rspec_metadata matcher will interpret the second string
+          # argument as a metadata symbol.
+          symbols.shift if symbols.first&.str_type?
```

Comment:
I first fixed this in the `rspec_metadata` matcher in `lib/rubocop/cop/rspec/mixin/metadata.rb` (see individual commits), but this made one test for the `RSpec/SortMetadata` cop fail. Looking at that cop, it seems to respect many more interesting cases, such as dynamic arguments, which could probably create more false positives in the `MetadataStyle` cop.

Example:

```ruby
context "something dynamic based on", variable do
  # ...
end
```

I decided not to dive into these cases and leave it to the AST matcher experts. (They may also be outside the scope of this PR, as this cop didn't worry about them so far.) Feel free to amend this PR if you see an obvious improvement.

Reply:
I suspect the real solution in mixin/metadata.rb would be to separate which metadata can be passed to `#Examples.all` vs `#ExampleGroups.all`, but I’m not sure I want to go down that rabbit hole 😅 The suggested fix works for me, too.